universal-application-tool-0.0.1/app/services/program/ProgramQuestionDefinition.java

```diff
@@ -0,0 +1,42 @@
+package services.program;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.auto.value.AutoValue;
+import java.util.Optional;
+import services.question.QuestionDefinition;
+
+/**
+ * {@link QuestionDefinition}s will not be stored in the database as part of the {@link
+ * models.Program} model. Only the question id will be stored. It is up the {@link ProgramService}
+ * to properly populate the question definition.
+ */
+@AutoValue
+public abstract class ProgramQuestionDefinition {
+
+  @JsonProperty("id")
+  public abstract long id();
+
+  @JsonIgnore
+  abstract Optional<QuestionDefinition> questionDefinition();
+
+  @JsonIgnore
+  public QuestionDefinition getQuestionDefinition() {
+    return questionDefinition().get();
+  }
+
+  public boolean hasQuestionDefinition() {
```

Comment:
question: just to double-check, we don't need `@JsonIgnore` here since it's not a getter?

Reply:
yeah. i think it only needs to be used on getters and setters.